### DISCUSSION TITLE
Dokku in docker

### DISCUSSION THREAD

**Discussion by krokhale** [dokku/dokku#4446] [Q&A]
I have looked up the history of the issues that led to this feature and I have been following this guide and it's working out great:

   https://dokku.com/docs/getting-started/install/docker/

I am running it locally on my Mini M1, and ports bind to 0.0.0.0. I have also added my ssh key inside the dokku container. I am having these issues:

- How does one push an app to this container, I have tried all sorts of combinations like:
    git remote add dokku dokku@0.0.0.0:3022:myapp (Is this the correct way to set the remote?)

- Also, can I not have it bind to 0.0.0.0? Can I run it in bridge mode, so that I have a specific IP to work with.

My aim is to run dokku locally just as I do on my Digital Ocean instance.

Appreciate the help!

**Comment by krokhale** [comment]
Also, I get asked the password in spite of adding the user ssh key inside the dokku container.

**Reply by josegonzalez** [reply]
When do you get asked for your password? You said you weren't able to push applications, so I'd assume this is happening elsewhere?

**Reply by krokhale** [reply]
@josegonzalez password while pushing my app to dokku. I figured it out, just had to enable ssh login inside dokku.

So here's what I have been playing around with: I am trying to have dokku run locally on my M1 machine, I also have access to a static IP address.

And I have been able to almost get everything done except for the part where dokku does not support ARM. So for instance If I were to push a simple express app to dokku, the dokku instance uses the system docker (M1 ARM docker) and tries to create a container with the defaults which happen to be v8/arm platform. BUT it seems like it should be pretty simple if dokku were to create a container with --platform=x86 option OR give the user an option to set the platform (Not sure if this already exists!). That would let me use M1 mini as a server.

Nothing super urgent, but just thought it might be interesting to flesh it out here!

Thanks!

**Reply by krokhale** [reply]
Also, thanks for updating the readme in the latest version.
